eclair-core/src/test/scala/fr/acinq/eclair/router/GraphSpec.scala

```diff
@@ -261,4 +262,89 @@ class GraphSpec extends AnyFunSuite {
       BlockHeight(714930), _ => true, includeLocalChannelCost = true)
     assert(path.path == Seq(edgeAB, edgeBC, edgeCE))
   }
+
+  test("fees less along C->D->E than C->E") {
+    /*
+    The channel C -> D is large enough for the payment and associated fees to go through, but C -> E is not.
+    A --> B --> C <-> D
+                 \   /
+                  \ /
+                   E
+    */
+    val edgeAB = makeEdge(1L, a, b, 10001 msat, 0, capacity = 200000 sat)
+    val edgeBC = makeEdge(2L, b, c, 10000 msat, 0, capacity = 200000 sat)
+    val edgeCD = makeEdge(3L, c, d, 10001 msat, 0, capacity = 200000 sat)
+    val edgeDC = makeEdge(4L, d, c, 10 msat, 0, capacity = 200000 sat)
+    val edgeCE = makeEdge(5L, c, e, 10003 msat, 0, capacity = 200000 sat)
+    val edgeDE = makeEdge(6L, d, e, 1 msat, 0, capacity = 200000 sat)
+    val graph = DirectedGraph(Seq(edgeAB, edgeBC, edgeCD, edgeDC, edgeCE, edgeDE))
+
+    val paths = yenKshortestPaths(graph, a, e, 90000000 msat,
+      Set.empty, Set.empty, Set.empty, 2,
+      Left(WeightRatios(1, 0, 0, 0, RelayFees(0 msat, 0))),
+      BlockHeight(714930), _ => true, includeLocalChannelCost = true)
+
+    assert(paths.length == 2)
+    assert(paths.head.path == Seq(edgeAB, edgeBC, edgeCD, edgeDE))
```

Comment:
```suggestion
    assert(paths.(0).path == Seq(edgeAB, edgeBC, edgeCD, edgeDE))
```
to keep it consistent with the next line.

Reply:
Updated it to use the index for consistency. I had changed it to head only because Intellij had colored it yellow.